### PR TITLE
kindsys: Adapt to new flattened lineage syntax in Thema

### PR DESCRIPTION
**Pull request by sdboyer**
**What is this feature?**

This updates all lineages in grafana/grafana to Thema's new syntax for defining lineages, introduced in grafana/thema#82.

**Why do we need this feature?**

Some more details are given on the upstream PR, but three key points:

* The syntax is a significant usability improvement (no more double-nested arrays, less terminology, etc.)
* Before this change, Thema simply did not support minor version downgrade lenses
* This change in Thema involved a lot of internal refactoring, paving the way for critical capabilities, like references between kinds.

**Who is this feature for?**

Anyone writing a kind in grafana.

**Special notes for your reviewer:**

This is scoped down as far as i'm able, but it's still pretty hefty. Several follow-up PRs are going to be necessary.

There are also some outstanding issues that probably need to be addressed before it can be merged:

- [x] `common.DataQuery` is inlined, but it should be `extends DataQuery` - [more details here](https://github.com/grafana/grafana/pull/68034/files#r1187781153). @spinillos is on this one

Deferred to a follow up are the following:

- [ ] A couple `DataQuery` composable kinds do not have any actual fields in them - only definitions. This has to change, but will take some doing. See [this comment](https://github.com/grafana/grafana/pull/68034#issuecomment-1540010612).
- [ ] Actually use `joinSchema`, e.g. to inject `common.DataQuery` (depends on the previous item)
- [x] Something in the the elasticsearch lineage [causes the CUE evaluator to hang](https://github.com/grafana/grafana/pull/68034/files#r1187800059). (#69182)

### CODE REVIEW

File: public/app/plugins/datasource/phlare/dataquery.gen.ts
Comment (sdboyer) [inline public/app/plugins/datasource/phlare/dataquery.gen.ts:26]: cuetsy is no longer noticing that `common.DataQuery` should be expressed via `extends` statement. Instead, it's inlining all the fields directly into the generated types. This requires a fix in cuetsy.

I'm fairly sure this is happening because of changes in [the CUE value actually being passed to cuetsy](https://github.com/grafana/thema/blob/main/lineage.cue#L258) - passing a reference to a field (`schema`) that's unified with a user-provided struct literal. Will need digging to figure it out, though.

File: pkg/plugins/pfs/pfs.go
Comment (sdboyer) [inline pkg/plugins/pfs/pfs.go:214]: This is quite baffling.

This unification - which should be a redundant no-op for core plugins when they have [this line](https://github.com/grafana/grafana/pull/68034/files#diff-3be98bfef812f58d55f118e27326f4130a284fd285d2a192a7435204caef2de1R23) - causes the CUE evaluator to hang. And _only_ for the elasticsearch composable kind.

i could almost understand this happening for Cloudwatch, which contains a recursive type. I wonder if there's some bizarre incidental shadowing problem going on. Either way, this deserves further investigation - but we can do it in a follow-up.

**Comment by sdboyer** [comment]
github's UI is broken and isn't letting me make the following comment on [this line](https://github.com/grafana/grafana/pull/68034/files#diff-6c6d5b0e7cec50f6f413b631ec41432efc11e7ea32ec09117e607540eee0ff12R31)

--

This schema doesn't contain any actual fields - just definitions. Really, this is my fault - i didn't realize we weren't guarding against this programatically already in Thema, and we didn't include as part of our instructions for schema authoring on schema day that schemas _must_ contain actual fields.

Thema now contains - or contained, as i've [temporarily reverted the check](https://github.com/grafana/thema/commit/53a6fb01cf3135287b2501a08b9381b5c9831bbd) in order to defer this to a follow-up - checks to enforce that schemas may not be empty. We'll need to refactor these schemas a bit to make sense in that context. To do so, we'll probably have to resolve deeper questions about what the DataQuery schema interface is actually trying to say - #60680

**Comment by sdboyer** [comment]
unmarking this as draft, it's ready for review modulo the known issue about inlined `common.DataQuery`.

**Review by aangelisc — APPROVED** [review]
👍 for Azure changes 😊

**Review by marefr — APPROVED** [review]
LGTM, just some lint and test failures left to resolve

**Comment by sdboyer** [comment]
OK, after some discussion in slack, we _will_ merge this without having fixed the `extends` issue, and instead defer that to a follow-up with the other issues.

**Comment by sdboyer** [comment]
Turns out we can't merge without a fix - inlining the type means we're no longer taking advantage of the hack-fix we have in the veneer for `DataQuery.datasource`, resulting in [tsc linter/compiler failures](https://drone.grafana.net/grafana/grafana/116627/2/5).

On the one hand, really need to get rid of that veneer hack. But for now, we're stuck with it, so we need the fix in cuetsy to stop inlining.

**Pull request merged**
